linera-base/src/crypto/mod.rs

```diff
@@ -135,25 +263,31 @@ impl FromStr for AccountPublicKey {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let value = hex::decode(s)?;
-        Ok(AccountPublicKey((value.as_slice()).try_into()?))
+        AccountPublicKey::from_slice(value.as_slice())
     }
 }
 
 impl Display for AccountPublicKey {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        self.0.fmt(f)
+        write!(f, "{}", hex::encode(self.as_bytes()))
```

Comment:
We could probably still do this, to keep the formatting parameters:

```suggestion
        hex::encode(self.as_bytes()).fmt(f)
```